dmutils/email/dm_mailchimp.py

```diff
@@ -45,20 +48,22 @@ def wrapper(*args, **kwargs):
 
     def create_campaign(self, campaign_data):
         try:
-            campaign = self._client.campaigns.create(campaign_data)
+            with log_external_request(service='Mailchimp'):
+                campaign = self._client.campaigns.create(campaign_data)
             return campaign['id']
         except RequestException as e:
             self.logger.error(
-                "Mailchimp failed to create campaign for '{0}'".format(
-                    campaign_data.get("settings").get("title")
+                "Mailchimp failed to create campaign for '{campaign_title}'".format(
+                    campaign_title=campaign_data.get("settings").get("title")
```

Comment:
If you want to chain `get` calls then (unless I've misremembered) you have to make sure the default value is a dictionary, so in this case you need `...get("settings", {}).get(...`

Reply:
I mean I didn't change that bit, so it clearly works as-is... But yeah that's better practice. I'll update it if you promise to finish reviewing so this can go in. ;)

Reply:
On this subject I saw something interesting the other day https://sedimental.org/ https://github.com/mahmoud/glom